### PR TITLE
GPII-2668:  Windows Process Reporter: C# "process not running" exception

### PR DESCRIPTION
**Pull request by klown**
Modified the C# ("dotNetProcesses.csx") to handle the exception where the process has exited while fetching information about it, specifically its process ID.

### CODE REVIEW

**Comment by klown** [comment]
ok to test

**Comment by klown** [comment]
@amb26 Since you so kindly reviewed the original ProcessReporter pull requests, could you have a look at this, when/if you have time?  Thanks!

File: gpii/node_modules/processReporter/dotNetProcesses.csx [hunk 1]
```diff
@@ -50,68 +47,77 @@ public class Startup
...
             foreach (ManagementObject p in processes) {
-                result.Add(makeProcInfo(p));
+                makeAndApprendProcInfo(p, result);
             }
         }
         return result.ToArray();
     }
 
-    public static ProcInfo makeProcInfo(ManagementObject process) {
+    public static void makeAndApprendProcInfo(ManagementObject processMO, ArrayList procInfos) {
```
Comment (stegru) [inline gpii/node_modules/processReporter/dotNetProcesses.csx]: app**r**end?

File: gpii/node_modules/processReporter/dotNetProcesses.csx [hunk 2]
```diff
@@ -50,68 +47,77 @@ public class Startup
...
-            case ThreadState.Standby:
-            case ThreadState.Transition:
-                procInfo.state = "Sleeping";
-                break;
+                case ThreadState.Initialized:
+                case ThreadState.Ready:
+                case ThreadState.Standby:
+                case ThreadState.Transition:
+                    procInfo.state = "Sleeping";
+                    break;
```
Comment (stegru) [inline gpii/node_modules/processReporter/dotNetProcesses.csx]: Some tab characters, before "case" on 106-111

**Comment by stegru** [comment]
Looks good to me.

I couldn't reproduce this naturally, so I changed the code to `Process.GetProcessById(pid + 1)`, and it no longer crashed - and isn't in the list as expected.

File: gpii/node_modules/processReporter/dotNetProcesses.csx (hunk 1 above)
Comment (amb26) [reply gpii/node_modules/processReporter/dotNetProcesses.csx]: Hitler apprende
https://www.youtube.com/watch?v=pQD9e9AoV3c

Comment (klown) [reply gpii/node_modules/processReporter/dotNetProcesses.csx]: At least the error was consistent.  Copy/paste is your friend.  Or, enemy.

**Comment by klown** [comment]
@stegru wrote:

> I couldn't reproduce this naturally, so I changed the code to Process.GetProcessById(pid + 1), and it no longer crashed - and isn't in the list as expected.

Yes, I couldn't make the exception happen.  The only reason I found out about it was CI, and, even there, it failed on two occasions.

File: gpii/node_modules/processReporter/dotNetProcesses.csx (hunk 2 above)
Comment (kaspermarkus) [reply gpii/node_modules/processReporter/dotNetProcesses.csx]: Didn't get these tabs on checkout of this code, but it could be my editor being nice

**Comment by kaspermarkus** [comment]
I was able to reproduce the issue quite consistently (every 3rd login or so) as described here: https://issues.gpii.net/browse/GPII-2777 .... It disappeared once I merge with this branch.

**Comment by klown** [comment]
@kaspermarkus I had a chat with @stegru in #fluid-tech, and he is satisfied with code, but he doesn't have the authority to merge it into master:

> [stegru>   ok, well, you'll have to ask one of the gatekeepers to merge it in - kasparnet is obviously happy with it because he's used it](https://botbot.me/freenode/fluid-tech/2018-01-16/?msg=95823399&page=1)

**Pull request merged**